### PR TITLE
MERGE GITHOOK COMMIT-MSG:

### LINKED ISSUE #3: Githook: Force commit norm
**We need to implement a script to check that the commit follows the code of conduct described in the wiki.
Then we need to deploy the script in .git/hooks/commit-msg**

### PR DESCRIPTION
**Pull request by AdrianWeaver**
This is the needed script in order to check the norm of your commits.

I did something fancy with regex to learn them a bit, the script is in bash.

Please make sure to chmod 755 .githooks/commit-msg

Close #3

### CODE REVIEW

**Comment by AdrianWeaver** [comment]
I suggest validating this as soon as possible to check for commits norms on the others :)

**Comment by AdrianWeaver** [comment]
If you have ideas on how to break the code I suggest the following procedure:
create a file, write the msg you want to try.
launch ``bash ./githooks/commit-msg filename``

**Comment by AdrianWeaver** [comment]
I added two commits to fix some cases I did not thing of yesterday.
The rest will need to be improved based on your usage of the script.

Please review and merge asap to help with the overall workflow 
@apayetRun @Elena-Lou @nicolkaver  @Keeghaan

**Comment by Elena-Lou** [comment]
"UPDATE && fix"  is successful

**Comment by Keeghaan** [comment]
It tells me to add ':' at the end of the title and when I do it tells me I didnt provide enough details.

**Comment by AdrianWeaver** [comment]
Can you provide a more accurate description ?
What did you test?
What was expected?
What did you get in return?


Example:
I tested : 
```ADD THIS FILE:

This is a description```

I expected the script to return `0`
The script sent an error :
```copy-paste the error here```

This way I can reproduce and find out if it's indeed an error or if it's intended.

**Comment by Keeghaan** [comment]
Tested:

UPDATE SIZES PCX TO VW BLA BLA B:

test

error:
words in commit: 1
Please provide a more detailed description

File: .githooks/commit-msg [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	if [[ $(cat $1 | grep -v '^\s*#' | sed -n '3,$p' | wc -w) -lt 2 ]]
+	then
+		echo "Please provide a more detailed description"
+		return 1
+	fi
+	line_nb=$(cat $1 | wc -l)
+	for i in $line_nb
+	do
+		line=$(cat $1 | sed -n "${i}p")
+		if [[ line =~ From\s\w{1,}repo\s\w{1,} ]]
```
Comment (Keeghaan) [inline .githooks/commit-msg:121]: Already noticed but still there is a small error here

File: .githooks/commit-msg [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+		then
+			echo "${ORANGE}Please note there is no need to add this line:${NC}"
+			echo $line
+		elif [[ line =~ On\sbranch\s\w*\/?\w* ]]
```
Comment (Keeghaan) [inline .githooks/commit-msg:125]: Same here.

**Review by Keeghaan — APPROVED** [review]
There is an error in the file, as mentioned in the comments.
Though it's not such a big deal so I am choosing to approve and you can fix it later.

**Review by Elena-Lou — APPROVED** [review]
Thank you for the great work

**Pull request merged**